src/main/java/uk/gov/hmcts/reform/ccd/document/am/service/impl/DocumentManagementServiceImpl.java

```diff
@@ -41,17 +47,32 @@ public ResponseEntity getDocumentMetadata(UUID documentId) {
     }
 
     @Override
-    public String extractDocumentMetadata(StoredDocumentHalResource storedDocument) {
+    public String extractCaseIdFromMetadata(Object storedDocument) {
+        if (storedDocument instanceof StoredDocumentHalResource) {
+            Map<String,String> metadata = ((StoredDocumentHalResource) storedDocument).getMetadata();
+            return metadata.get("caseId");
+        }
         return null;
     }
 
     @Override
-    public ResponseEntity<?> getDocumentBinaryContent(UUID documentId) {
-        return null;
+    public ResponseEntity<Resource> getDocumentBinaryContent(UUID documentId) {
+
+        try  {
+            return documentStoreFeignClient.getDocumentBinary(documentId);
+
+        } catch (FeignException ex) {
+            log.error("Cannot download document that is stored::" + ex.status());
+            throw new ResourceNotFoundException("Cannot download document that is stored");
```

Comment:
404 is not a correct return code for all the errors. It could be 400, 403 as well

Reply:
I am not getting.Need Clarity.